store/storage/rocksdb/batch.go

```diff
@@ -0,0 +1,59 @@
+package rocksdb
+
+import (
+	"encoding/binary"
+
+	"github.com/linxGnu/grocksdb"
+)
+
+type Batch struct {
+	version  uint64
+	ts       [TimestampSize]byte
+	storage  *grocksdb.DB
+	cfHandle *grocksdb.ColumnFamilyHandle
+	batch    *grocksdb.WriteBatch
+}
+
+// NewBatch creates a new versioned batch used for batch writes. The caller
+// must ensure to call Write() on the returned batch to commit the changes and to
+// destroy the batch when done.
+func NewBatch(db *Database, version uint64) Batch {
+	var ts [TimestampSize]byte
+	binary.LittleEndian.PutUint64(ts[:], uint64(version))
```

Comment:
To aid future you and other maintainers
```go
         var versionAsTimestamp [TimestampSize]byte
	binary.LittleEndian.PutUint64(versionAsTimestamp[:], uint64(version))
```

Reply:
I personally don't think this buys much IMO

Reply:
It's readability and making choices of version for the timestamp very obvious, but it isn't the end of the world :-)